File: simple/java/move-method_isolate-switch_after.java
class Order {
  // ...

  public double calculateTotal() {
    // ...
    total = Discounts.applyRegionalDiscounts($total);
    total = Discounts.applyCoupons($total);
    // ...
  }


class Discounts {
  // ...

  public static double applyRegionalDiscounts(total) {
    double result = total;
    switch (user.getCountry()) {
      case "US": result *= 0.85; break;
      case "RU": result *= 0.75; break;
      case "CN": result *= 0.9; break;
      // ...
    }
    return result;
  }

  public static double applyCoupons(total) {
      // ...
  }

File: simple/typescript/move-method_isolate-switch_after.ts
class Order {
  // ...

  calculateTotal(): number {
    // ...
    total = Discounts.applyRegionalDiscounts(total, user.getCountry());
    total = Discounts.applyCoupons(total);
    // ...
  }


class Discounts {
  // ...

  static applyRegionalDiscounts(total: number, country: string): number {
    let result = total;
    switch (country) {
      case "US": result *= 0.85; break;
      case "RU": result *= 0.75; break;
      case "CN": result *= 0.9; break;
      // ...
    }
    return result;
  }

  static applyCoupons(total: number): number {
      // ...
  }
